src/flint/fmpz.jl

```diff
@@ -2659,6 +2659,94 @@ Base.promote_rule(::Type{fmpz}, ::Type{T}) where {T <: Integer} = fmpz
 
 promote_rule(::Type{fmpz}, ::Type{T}) where {T <: Integer} = fmpz
 
+###############################################################################
+#
+#  Perfect power detection
+#
+###############################################################################
+
+# 1, 0, -1 are perfect powers
+# ex is not guarenteed to be maximal
+function _is_perfect_power(a::fmpz)
+  rt = fmpz()
+  ex = ccall((:fmpz_is_perfect_power, libflint), Int, (Ref{fmpz}, Ref{fmpz}), rt, a)
+  return rt, ex
+end
+
+@doc Markdown.doc"""
+    is_perfect_power(a::IntegerUnion)
+
+Returns whether $a$ is a perfect power, that is, whether $a = m^r$ for some
+integer $m$ and $r > 1$.
+"""
+function is_perfect_power(a::fmpz)
+  _, ex = _is_perfect_power(a)
+  return ex > 0
+end
+
+is_perfect_power(a::Integer) = is_perfect_power(fmpz(a))
+
+# Returns $e$, $r$ such that $a = r^e$ with $e$ maximal. Note: $1 = 1^0$.
+function _maximal_integer_root(a::fmpz)
```

Comment:
I would like to expose this to the user, but I don't know a good name. So I postpone this to another PR.

Reply:
I don't understand. What are you saving for another PR. If the name is `is_perfect_power`, I would be surprised if it returned anything other than a bool.

Reply:
I just copied it from Hecke. I save renaming this function to `fxn1` and exporting it to another PR.